### PR TITLE
Implemented Fixed length random generator

### PR DESCRIPTION
**Pull request by santhoshTpixler**
**Implements #245** 

As well as few improvements : 
* Removed duplicate testCaseTokens for efficiency
* Every occurrence of a RANDOM.NUMBER will be
  replaced with a unique number

Tests are yet to be added. **_Looking for suggestion/improvements._**

### CODE REVIEW

**Comment by jneate** [comment]
@santhoshTpixler looks good mate

Cheers for picking up the PR 👍 

@authorjapps @santhoshTpixler suggest that this PR and PR #273 be merged at the same time as they both benefit from the updated Wiki pages.

File: core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java [hunk 1]
```diff
@@ -16,5 +23,60 @@ public void testResolve_knownTokens() {
...
+    @Test
+    public void testFixedRandomGenerator_failure_min() {
+    	exceptionRule.expect(RuntimeException.class);
+    	FixedRandomGenerator.getGenerator(0).generateRandomNumber();
+    }
+    
+    @Test
+    public void testFixedRandomGenerator_failure_max() {
+    	exceptionRule.expect(RuntimeException.class);
+    	FixedRandomGenerator.getGenerator(20).generateRandomNumber();
```
Comment (authorjapps) [inline core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java]: 👍 
We can capture this into the Wiki page.

File: core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java [hunk 2]
```diff
@@ -16,5 +23,60 @@ public void testResolve_knownTokens() {
...
+    @Test
+    public void testFixedRandomUniqueness(){
+    	String result = resolveKnownTokens("${RANDOM.NUMBER:12},${RANDOM.NUMBER:12}");
+    	String[] split = result.split(",");
+    	assertTrue(split[0] != split[1]);
+    }
+    
+    
+}
+
```
Comment (authorjapps) [inline core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java]: Minor one - These empty line basically showing up here...
Can you please delete them

File: core/src/main/java/org/jsmart/zerocode/core/utils/FixedRandomGenerator.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package org.jsmart.zerocode.core.utils;
+
+import java.util.HashMap;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ *  A fixed length random generator supports up to 19 digits
+ * @author santhosh
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/utils/FixedRandomGenerator.java]: Can you remove the author please?
> Git keeps the history of authors.

File: core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java [hunk 4]
```diff
@@ -16,5 +23,60 @@ public void testResolve_knownTokens() {
...
     }
-}
\ No newline at end of file
+    
+    @Test
+    public void testFixedRandomTokenReplace() {
+    	 String clientId = "zerocode-clientid_${RANDOM.NUMBER:10}";
+         String uniqueClientId = resolveKnownTokens(clientId);
+         String uniqueId = uniqueClientId.substring("zerocode-clientid_".length());
+         assertTrue(uniqueId.length() == 10);
```
Comment (authorjapps) [inline core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java]: +1

File: core/src/main/java/org/jsmart/zerocode/core/utils/TokenUtils.java [hunk 5]
```diff
@@ -34,7 +37,12 @@ public static void populateParamMap(Map<String, Object> paramaMap, String runTim
         getKnownTokens().forEach(inStoreToken -> {
                     if (runTimeToken.startsWith(inStoreToken)) {
                         if (runTimeToken.startsWith(RANDOM_NUMBER)) {
-                            paramaMap.put(runTimeToken, System.currentTimeMillis() + "");
+                        	String[] slices = runTimeToken.split(":");
+                        	if (slices.length == 2) {
+                        		paramaMap.put(runTimeToken, FixedRandomGenerator.getGenerator(Integer.parseInt(slices[1])));
+                        	}else {
+								paramaMap.put(runTimeToken, RandomGenerator.getGenerator());
```
Comment (authorjapps) [inline core/src/main/java/org/jsmart/zerocode/core/utils/TokenUtils.java]: Basically, this calls the `toString()` of the generator?

**Comment by authorjapps** [comment]
> @authorjapps @santhoshTpixler suggest that this PR and PR #273 be merged at the same time as they both benefit from the updated Wiki pages.

Thanks. Understood.

**Comment by authorjapps** [comment]
PR looks good.
Added some very minor cosmetic comments, if you could take care please @santhoshTpixler .

_Sorry again for the late replies as we all were occupied a bit with other jobs._

File: core/src/main/java/org/jsmart/zerocode/core/utils/TokenUtils.java (hunk 5 above)
Comment (santhoshTpixler) [reply core/src/main/java/org/jsmart/zerocode/core/utils/TokenUtils.java]: Yeah, `toString()` is called for every instance of the token. This functionality helps us to produce different random numbers in the same step with the same length.

**Comment by authorjapps** [comment]
> This branch has conflicts that must be resolved
Use the web editor or the  to resolve conflicts.
Conflicting files
core/src/test/java/org/jsmart/zerocode/core/utils/TokenUtilsTest.java

When you get chance please @santhoshTpixler .

Also, Can you take care of these minor comments?

**Comment by santhoshTpixler** [comment]
I have made minor changes. There could be some formatting changes. I am not sure of the code style #280. So kindly take care of it.

**Pull request merged**